tests/pools/abstract/abstract-pool.test.js

```diff
@@ -61,4 +61,40 @@ describe('Abstract pool test suite', () => {
       const pool = new StubPoolWithIsMainMethod(1, '').toThrowError()
```

Comment:
This test doesn't look right to me :thinking:  
the `.toThrowError()` should be called on `expected` and not inside it

I could be wrong, but also I think just calling `.toThrowError()` with no `new Error(...)` argument always returns `true` :confused:  
This is really odd by mocha but I think it is this way :slightly_frowning_face: